src/main/java/com/exceptionless/exceptionlessclient/queue/DefaultEventQueue.java

```diff
@@ -158,6 +158,13 @@ private void processSubmissionResponse(
       return;
     }
 
+    if (response.isRateLimited()) {
```

Comment:
@niemyjski Are we sure that the rate limit header is not returned with a successful code? Because we are checking it after checking for success so it will not work as intended in that scenario.

Reply:
The header might not be there, I'd treat it as nullable.  Here are the two api calls it should be set on (https://github.com/exceptionless/Exceptionless.JavaScript/blob/feature/workspaces/packages/core/src/submission/SubmissionClientBase.ts#L29-L50)